### ISSUE TITLE
xLights: Sub-model Select Nodes window node range text box starts enabled then goes read-only

### ISSUE THREAD

**Issue by Holiday851** [xLightsSequencer/xLights#3145]
Issue: Node list text box starts off enabled, then becomes read-only after editing text and selecting another node.

Repro:
Select a custom model in Layout
Click in the sub-model text box ("Click to edit")
Click the "..." button to edit
Click "Add" to create a new sub-model
Double click on "Bottom" to open up "Select Nodes"
Ensure the "Ordered Selection" is checked.
Start double-clicking to add ordered nodes.
Click into the text box at the bottom of the screen.
Add a comma to end of the node list to add a skip.
Double-click some more nodes.
Try to click into the text box at the bottom of the screen to add another skip.
You cannot.
Requires user to either keep saving and then re-editing (quite a pain for places in the model where every-other pixel in an area is a skip), or editing it in the Nodes Ranges text box on the previous screen (which has its own issues - see issue 3144).

Expected:
Either the text box stays read-only for the whole time (so skips can be added via text), or it stays enabled for the whole time (if the former, then 3144 would  be nice to have a way to add skips).

**Comment by Holiday851** [comment]
Version 2022.05 64-bit Windows 10

**Comment by computergeek1507** [comment]
It is working as intended for me. Please make a video to help explain the issue.

![nodeskip](https://user-images.githubusercontent.com/2293065/156090326-76b32a3c-8673-4286-83b4-7554efe563fd.gif)

**Comment by Holiday851** [comment]
https://user-images.githubusercontent.com/95460274/156092147-4450c530-ffa3-4ab8-9b96-65bde938b715.mp4

In a video, it's hard to show that you cannot type into something since it doesn't do anything, but in your repro video, I don't see you ever type anything into the box - you just set the caret and then click on a node.

You have to select nodes, type (for example a comma), then select more nodes, then try the text box again.

**Comment by Holiday851** [comment]
[TextBoxGoesReadOnly.zip](https://github.com/smeighan/xLights/files/8158330/TextBoxGoesReadOnly.zip)

Not sure if it's just an issue with the video viewer or what, so I zipped it up and uploaded the zip.

Note: after I click on the 9 node, then I type a comma, then clilck on the other nodes, then try to click back in the edit box.

**Comment by Holiday851** [comment]
[TextBoxGoesReadOnlySteps.zip](https://github.com/smeighan/xLights/files/8158387/TextBoxGoesReadOnlySteps.zip)

Maybe this is more useful?

**Comment by Holiday851** [comment]
Do I have to do something to clear the Waiting For User Feedback flag? And
if so, how do I clear it (not experienced with GitHub and don't immediately
see a way to clear it)?

**Comment by keithsw1111** [comment]
Fundamentally your problem is you cant actually enter a comma when in this mode ... and entering a - actually turns off ordered selection. If you want to add commas for now you do them in the previous dialog. Adding a comma is likely to be tricky as this is not just operating as a plain old text box. Everytime you click on something xlights parses the string and recreates it.

Given this is perfectly doable on the previous dialog this does not seem to be a high priority change

**Comment by Holiday851** [comment]
Roger that. Later, I'll see about getting a video repro on the bug where
the previous screen doesn't allow you to scroll back to the beginning of
the text (when you caret into the text box and try to scroll left, it just
won't go to the beginning when you drag the Node Range column wider which
is necessary because it starts out so narrow, it's difficult to edit).

**Comment by cybercop23** [comment]
@Holiday851 can this be closed?

**Comment by Holiday851** [comment]
Ya, I have my own program to build sub-models, so you can close everything.
